ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/CacheStore.java

```diff
@@ -460,84 +464,84 @@ public final int removeWriteState (Uid objUid, int ft)
         int status = NOT_PRESENT;
         Object[] elements = null;
 
-        try
-        {
-            elements = getList(objUid).toArray();
-        }
-        catch (Exception ex)
-        {
-            elements = null;
+        LinkedList list = getList(objUid);
+        synchronized (list) {
+            try {
+                elements = list.toArray();
```

Comment:
Would not be better to check the "wrong" state in a different way than catching an `Exception`?

Reply:
I wanted to keep the patch as close to the original intention as I could. I do not actually expect an exception here anymore. What I saw was that the list through an array out of bounds exception but I believe that was because the list.toArray was happening concurrent with a list.addFirst (IIRC) - we could likely remove the catch entirely now.